tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1099,6 +1101,14 @@ ROOT::Experimental::Internal::RNTupleFileWriter::RFileProper::ReserveBlobKey(siz
       Write(localKeyBuffer, kBlobKeyLen, offsetKey);
    }
 
+   if (keyBlob.WasAllocatedInAFreeSlot()) {
+      // If the key was allocated in a free slot, the last 4 bytes of its buffer contain the new size
+      // of the remaining free slot and we need to write it to disk before the key gets destroyed at the end of the
+      // function.
+      auto nbytesFree = *(Int_t *)(keyBlob.GetBuffer() + nbytes);
```

Comment:
The pointer resulting from `keyBlob.GetBuffer() + nbytes` may not be aligned, which could break on some architectures. I think we must use `memcpy` (which I think also avoids any endianess problems)